gcc/rust/backend/rust-compile-pattern.cc

```diff
@@ -80,6 +80,20 @@ CompilePatternCaseLabelExpr::visit (HIR::WildcardPattern &pattern)
     = build_case_label (NULL_TREE, NULL_TREE, associated_case_label);
 }
 
+void
+CompilePatternCaseLabelExpr::visit (HIR::LiteralPattern &pattern)
+{
+  // Compile the literal
+  HIR::LiteralExpr *litexpr
+    = new HIR::LiteralExpr (pattern.get_pattern_mappings (),
+			    pattern.get_literal (), pattern.get_locus (),
+			    std::vector<AST::Attribute> ());
+
+  tree lit = CompileExpr::Compile (litexpr, ctx);
+
+  case_label_expr = build_case_label (lit, NULL_TREE, associated_case_label);
```

Comment:
Oh no way this is really cool, I didn't realise you could do this! But of course you can. I do all the name/type resolution work so that we could support matching complex tuples but doing the match on them really confused me but this opens it up.

In theory we should be able to match integers this way and the case labels support ranges here so that we can support matching integer ranges.